cmds/core/stat/run_darwin.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2025 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+	"time"
+)
+
+func fromTimespec(ts syscall.Timespec) string {
+	t := time.Unix(ts.Sec, ts.Nsec)
+	return t.Format("Jan  2 15:04:05 2006")
+}
+
+// device inode perm nlink uid gid rdev size atime mtime
+// ctime btime blksize blocks st_flags filename
+const defFormat = "%d %d %s %d %d %d %d %d %q %q %q %q %d %d %d %s\n"
+
+func run(stdout io.Writer, stderr io.Writer, files ...string) int {
+	var errCode int
+	for _, file := range files {
+		fi, err := os.Lstat(file)
+		if err != nil {
+			fmt.Fprintf(stderr, "stat: %v", err)
+			errCode = 1
+			continue
+		}
+
+		stat, ok := fi.Sys().(*syscall.Stat_t)
+		if !ok {
+			fmt.Fprintf(stderr, "stat: missing fileinfo")
+			errCode = 1
+			continue
+		}
+
+		fmt.Fprintf(stdout, defFormat,
+			stat.Dev,
+			stat.Ino,
+			fi.Mode().Perm(),
+			stat.Nlink,
+			stat.Uid,
+			stat.Gid,
+			stat.Rdev,
+			stat.Size,
+			fromTimespec(stat.Atimespec),
+			fromTimespec(stat.Mtimespec),
+			fromTimespec(stat.Ctimespec),
+			fromTimespec(stat.Birthtimespec),
+			stat.Blksize,
+			stat.Blocks,
+			stat.Flags,
+			fi.Name(),
+		)
```

Comment:
Also, we should discuss if we want to keep the same default format for the output. They seem to differ quite much:

coreutils stat:
```
$ stat ./stat_test.go 
  File: ./stat_test.go
  Size: 825       	Blocks: 8          IO Block: 4096   regular file
Device: 0,41	Inode: 40659960    Links: 1
Access: (0644/-rw-r--r--)  Uid: ( 1000/  leongross)   Gid: ( 1000/  leongross)
Context: unconfined_u:object_r:user_home_t:s0
Access: 2025-03-09 16:15:43.756728492 +0100
Modify: 2025-03-09 16:15:43.747728510 +0100
Change: 2025-03-09 16:15:43.747728510 +0100
Birth: 2025-03-09 16:15:43.747728510 +0100
```

vs our format:
```
$ ./stat ./stat_tet.go
41 40659960 -rw-r--r-- 1 1000 1000 0 825 "Mar  9 16:15:43 2025" "Mar  9 16:15:43 2025" "Mar  9 16:15:43 2025" 4096 8 stat_test.go
```

Reply:
this should be done by -x flag, `-x      Display information in a more verbose way as known from some Linux distributions.`

Reply:
Also format is heavily dependent on operating system, so my intention was to show what `stat` structure have for corresponding operating system.